src/cpu/core_dyn_x86/decoder.h

```diff
@@ -140,28 +140,32 @@ static Bit8u decode_fetchb(void) {
 	decode.code+=1;
 	return mem_readb(decode.code-1);
 }
-static Bit16u decode_fetchw(void) {
-	if (GCC_UNLIKELY(decode.page.index>=4095)) {
-   		Bit16u val=decode_fetchb();
-		val|=decode_fetchb() << 8;
+static uint16_t decode_fetchw()
+{
+	if (GCC_UNLIKELY(decode.page.index >= 4095)) {
+		uint16_t val = decode_fetchb();
+		val |= decode_fetchb() << 8;
 		return val;
 	}
-	*(Bit16u *)&decode.page.wmap[decode.page.index]+=0x0101;
```

Comment:
PPC: Fix needs to be aligned and native endian (not swapped to little-endian like the `host_*` functions do).